### PR TITLE
feat: Update doc and code for Maven plugin

### PR DESCRIPTION
**Pull request by laurentsimon**

### CODE REVIEW

File: experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java [hunk 1]
```diff
@@ -88,7 +89,7 @@ public void execute() throws MojoExecutionException, MojoFailureException {
                         element(name("groupId"), artifact.getGroupId()),
                         element(name("artifactId"), artifact.getArtifactId()),
                         element(name("version"), artifact.getVersion()),
-                        element(name("type"), "intoto.build.slsa"),
+                        element(name("type"), "build.slsa"),
```
Comment (laurentsimon) [inline experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:92]: @AdamKorcz please confirm this change is correct

File: experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java [hunk 2]
```diff
@@ -117,41 +118,36 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
-            } catch(MojoExecutionException e) {
-                // TODO: Properly interpret the output based on the verification plugin.
-                getLog().info("Skipping slsa verification: Fail to run slsa verifier.");
-                return;
-            }
+            // Run slsa verification on the artifact and print the result
+            // It will never fail the build process
+            // TODO(#665): Harden against script injections.
+            String arguments = "verify-artifact --provenance-path ";
+            arguments += "${project.build.directory}/slsa/" + artifact.getArtifactId() + "-" + artifact.getVersion() + "-jar.build.slsa ";
```
Comment (laurentsimon) [inline experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:129]: @AdamKorcz please confirm this change is correct (I removed the intoto part of the name)

File: experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java [hunk 3]
```diff
@@ -117,41 +118,36 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+            arguments += " --source-uri ./ ${project.build.directory}/slsa/" + artifact.getArtifactId() + "-" + artifact.getVersion() + ".jar";
+            // NOTE: This command may throw an exception, indicating that provenance verification failed.
```
Comment (laurentsimon) [inline experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:131]: @AdamKorcz please confirm this change is correct

File: experimental/maven-plugin/README.md [hunk 4]
```diff
@@ -11,9 +11,9 @@ The plugin wraps the [the slsa verifier](https://github.com/slsa-framework/slsa-
 
 ## Prerequisites
 
-To use the plugin you must have Go, Java and Maven installed. It has currently only been tested on Ubuntu.
```
Comment (laurentsimon) [inline experimental/maven-plugin/README.md:14]: @AdamKorcz please confirm this change is correct

File: experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java (hunk 1 above)
Comment (AdamKorcz) [reply experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:92]: It is correct.

File: experimental/maven-plugin/README.md (hunk 4 above)
Comment (AdamKorcz) [reply experimental/maven-plugin/README.md:14]: It is correct.

File: experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java (hunk 2 above)
Comment (AdamKorcz) [reply experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:127]: Did you also remove the `catch(`?

Comment (laurentsimon) [reply experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:127]: I did. Build will fail if verification fails. Something i dont fully understand yet, is where the expected repository comes from. I can add a comment

**Comment by laurentsimon** [comment]
@AdamKorcz friendly ping. Can you resolve the comments if my answers clarify things?

File: experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java (hunk 2 above)
Comment (AdamKorcz) [reply experimental/maven-plugin/src/main/java/io/github/slsa-framework/SlsaVerificationMojo.java:127]: LGTM

**Comment by AdamKorcz** [comment]
> @AdamKorcz friendly ping. Can you resolve the comments if my answers clarify things?

Sorry for the delay. Your changes LGTM!

**Comment by laurentsimon** [comment]
@kpk47 can you take a look?

**Comment by laurentsimon** [comment]
@kpk47 ptal

**Review by ianlewis — APPROVED** [review]

**Pull request merged**